src/main/webapp/js/Navigation.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Creates navigation depending on whether user is logged in or not
+ */
+
+import React, { useState, useEffect } from 'react';
+
+function Navigation() {
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [user, setUser] = useState("");
```

Comment:
Let's rename this to username and setUsername.

Also, store the default value as null. Then you can remove loggedIn and isLoggedIn because you can just check whether username is null:

```
const [username, setUsername] = setState(null);
const loggedIn = !(username === null);
```

Reply:
Yeah good idea, thanks! What do you think about checking (username !== null) in the return statement instead? It would be a bit less clear than having a loggedIn variable, but that'd be one less call when setUsername is called???

Reply:
It's fine.

What do you mean by 

> but that'd be one less call when setUsername is called???

Also you left in a doublequote in your newest commit

Reply:
I was thinking having a loggedIn variable would be an extra place where username is used and called, so when the component renders and username is fetched, then loggedIn would have to be updated then rendered too. Having no loggedIn variable and just doing the logic in the return statement would be one less call i think?? Trivial but was wondering if I could leave the implementation like that